src/components/Select.tsx

```diff
@@ -1,28 +1,31 @@
 import './styles/select.less'
 
 import React from 'react'
-import PropTypes from 'prop-types'
 import classNames from 'classnames'
 
-/**
- * @typedef {Object} Option
- * @property {string | number | undefined} [key]
- * @property {string | number} [value]
- * @property {string | number | undefined} [text]
- * @property {boolean | undefined} [disabled]
- */
+interface Option {
+    /** Optional key for each option. If not provided `value` is used. */
+    key?: string | number
+    /** Value of the option. */
+    value: string | number
+    /** Text to display for the option. */
+    text?: string | number
+    /** Whether the options is disabled or not. */
+    disabled?: boolean
```

Comment:
Also ref.: https://twist.com/a/1585/ch/190200/t/1504732/c/58090586

(not sure if possible, but worth discussing a better solution)

Reply:
Have moved everything to default values here, I'll try to do the same to other components. I'm not sure this is a good rule in every case, especially when string/number fields are involved (is ""/0 an acceptable default value?), so there is some nuance.

Reply:
What I suggested is only true of empty collections and booleans. For other types it's only good if reasonable default can be found.

Reply:
This plays hell on tests as well as now we'll have to provide all the non-optional props. Semantically I feel this is wrong too as the component has this as an optional prop (as it has a default value set internally). So the callee Props interface does have this as optional.

Modelling this as non-optional is probably wrong, I've reverted this change in this PR.

Reply:
Sad but true :/

Reply:
Resolved as per here: https://twist.com/a/1585/ch/190200/t/1504732/c/58095833